src/render/draw_fill.js

```diff
@@ -60,11 +54,11 @@ function drawFill(painter: Painter, sourceCache: SourceCache, layer: FillStyleLa
 }
 
 function drawFillTiles(painter, sourceCache, layer, coords, drawFn) {
-    if (isPatternMissing(layer.paint.get('fill-pattern'), painter)) return;
-
     let firstTile = true;
     for (const coord of coords) {
         const tile = sourceCache.getTile(coord);
+        if (layer.paint.get('fill-pattern').constantOr((1: any)) && !tile.patternsLoaded()) continue;
```

Comment:
Just confirming: with this nice & clean patternsLoaded() check, we're no longer checking whether the particular patterns needed by the fill-pattern value are present in the image atlas -- is that okay?